pkg/tinydtls/include/sock_dtls_types.h

```diff
@@ -74,9 +75,12 @@ struct sock_dtls {
         size_t datalen;                     /**< data length */
         session_t *session;                 /**< Session information */
     } buffer;
-    credman_tag_t tag;                      /**< Credential tag of a registered
-                                                (D)TLS credential */
+    char psk_hint[CONFIG_DTLS_PSK_ID_HINT_MAX_SIZE]; /**< PSK Identity hint */
+    credman_tag_t tags[CONFIG_DTLS_CREDENTIALS_MAX]; /**< Tags of the available credentials */
+    unsigned tags_len;                      /**< Number of tags in the list 'tags' */
```

Comment:
I know that this would probably mean a redesign of the `credman` API so I actually would prefer that to be a follow-up, if that is the case, but: Would it make sense to have `tags` (and also maybe `psk_hint`) be a linked list instead of an array?

Reply:
> I know that this would probably mean a redesign of the `credman` API so I actually would prefer that to be a follow-up, if that is the case, but: Would it make sense to have `tags` (and also maybe `psk_hint`) be a linked list instead of an array?

Not sure if that would actually affect `credman`'s API. The `psk_hint` is just a string, so it should stay as is. It seems a bit more cumbersome from the user perspective having to allocate each list element to register a new tag.